beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1559,128 +1597,154 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
     /// Execute the fork choice algorithm and enthrone the result as the canonical head.
     pub fn fork_choice(&self) -> Result<(), Error> {
         metrics::inc_counter(&metrics::FORK_CHOICE_REQUESTS);
+        let overall_timer = metrics::start_timer(&metrics::FORK_CHOICE_TIMES);
 
-        // Start fork choice metrics timer.
-        let timer = metrics::start_timer(&metrics::FORK_CHOICE_TIMES);
+        let result = self.fork_choice_internal();
 
-        // Determine the root of the block that is the head of the chain.
-        let beacon_block_root = self.fork_choice.find_head(&self)?;
+        if result.is_err() {
+            metrics::inc_counter(&metrics::FORK_CHOICE_ERRORS);
+        }
 
-        // If a new head was chosen.
-        let result = if beacon_block_root != self.head_info()?.block_root {
-            metrics::inc_counter(&metrics::FORK_CHOICE_CHANGED_HEAD);
+        metrics::stop_timer(overall_timer);
 
-            let beacon_block = self
-                .get_block(&beacon_block_root)?
-                .ok_or_else(|| Error::MissingBeaconBlock(beacon_block_root))?;
+        result
+    }
 
-            let beacon_state_root = beacon_block.state_root();
-            let beacon_state: BeaconState<T::EthSpec> = self
-                .get_state(&beacon_state_root, Some(beacon_block.slot()))?
-                .ok_or_else(|| Error::MissingBeaconState(beacon_state_root))?;
+    fn fork_choice_internal(&self) -> Result<(), Error> {
```

Comment:
There was some pretty silly indentation going on in the old `fork_choice` function, so I split it into `fn fork_choice` and `fn fork_choice_internal` and cleaned it out. Sorry about the noisy diff.